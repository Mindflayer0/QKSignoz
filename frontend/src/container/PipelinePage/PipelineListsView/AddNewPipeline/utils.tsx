import React from 'react';

import { pipelineFields } from '../config';
import DescriptionTextArea from './FormFields/DescriptionTextArea';
import FilterSearch from './FormFields/FilterSearch';
import NameInput from './FormFields/NameInput';
// import ProcessorTags from './FormFields/ProcessorTags';

export const renderPipelineForm = (): // setTagsListData: (tags: Array<string>) => void,
// tagsListData?: Array<string>,
Array<JSX.Element> =>
	pipelineFields.map((fieldName) => {
		if (fieldName.id === '1') {
			return <FilterSearch key={fieldName.id} fieldData={fieldName} />;
		}
		if (fieldName.id === '2') {
			return <NameInput key={fieldName.id} fieldData={fieldName} />;
		}
		/* if (fieldName.id === '3') {
			return (
				<ProcessorTags
					key={fieldName.id}
					fieldData={fieldName}
					setTagsListData={setTagsListData}
					tagsListData={tagsListData || []}
				/>
			);
		} */
		return <DescriptionTextArea key={fieldName.id} fieldData={fieldName} />;
	});
