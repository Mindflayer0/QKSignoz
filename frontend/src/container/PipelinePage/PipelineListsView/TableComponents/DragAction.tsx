import { Switch } from 'antd';
import React from 'react';

import { HolderOutlinedIcon, LastActionColumn } from '../styles';

function DragAction(): JSX.Element {
	return (
		<LastActionColumn>
			<span>
				<Switch />
			</span>
			<span style={{ cursor: 'move' }}>
				<HolderOutlinedIcon />
			</span>
		</LastActionColumn>
	);
}

export default DragAction;
