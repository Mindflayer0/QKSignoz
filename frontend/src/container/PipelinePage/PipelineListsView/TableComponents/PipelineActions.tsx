import { Space } from 'antd';
import React from 'react';

import { DeleteFilledIcon, EditOutlinedIcon, EyeFilledIcon } from '../styles';

function PipelineActions({
	editAction,
	deleteAction,
}: PipelineActionsProps): React.ReactElement {
	return (
		<Space size="middle">
			<span>
				<EditOutlinedIcon onClick={editAction} />
			</span>
			<span>
				<EyeFilledIcon />
			</span>
			<span>
				<DeleteFilledIcon onClick={deleteAction} />
			</span>
		</Space>
	);
}

interface PipelineActionsProps {
	editAction: VoidFunction;
	deleteAction: VoidFunction;
}
export default PipelineActions;
