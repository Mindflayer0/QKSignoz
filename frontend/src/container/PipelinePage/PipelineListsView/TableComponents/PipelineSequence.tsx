import { Avatar } from 'antd';
import { themeColors } from 'constants/theme';
import React from 'react';

function PipelineSequence({
	value,
}: PipelineSequenceProps): React.ReactElement {
	return (
		<Avatar style={{ background: themeColors.navyBlue }} size="small">
			{value}
		</Avatar>
	);
}

interface PipelineSequenceProps {
	value: number;
}
export default PipelineSequence;
