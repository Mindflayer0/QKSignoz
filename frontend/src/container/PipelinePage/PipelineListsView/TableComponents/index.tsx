import React from 'react';

import { ActionBy, PipelineColumn } from '..';
import { ListDataStyle, ProcessorIndexIcon } from '../styles';
import PipelineSequence from './PipelineSequence';
import Tags from './Tags';

function TableComponents({
	columnKey,
	record,
}: TableComponentsProps): JSX.Element {
	if (columnKey === 'orderid') {
		return <PipelineSequence value={record} />;
	}
	if (columnKey === 'tags') {
		return <Tags tags={record} />;
	}
	if (columnKey === 'updatedBy') {
		return <span>{record.username}</span>;
	}
	if (columnKey === 'id') {
		return (
			<ProcessorIndexIcon size="small">{Number(record) + 1}</ProcessorIndexIcon>
		);
	}
	if (columnKey === 'text') {
		return <ListDataStyle>{record}</ListDataStyle>;
	}
	return <span>{record}</span>;
}

interface TableComponentsProps {
	columnKey: string;
	record: ActionBy & PipelineColumn['orderid'] & PipelineColumn['tags'];
}

export default TableComponents;
