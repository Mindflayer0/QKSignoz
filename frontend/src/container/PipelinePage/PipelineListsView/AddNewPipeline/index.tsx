import { Button, Divider, Form, Modal } from 'antd';
import React, { useCallback, useEffect, useMemo } from 'react';
import { useTranslation } from 'react-i18next';
import { useSelector } from 'react-redux';
import { AppState } from 'store/reducers';
import { PipelineData } from 'types/api/pipeline/def';
import AppReducer from 'types/reducer/app';

import { ActionMode, ActionType } from '../../Layouts/Pipeline';
import { ModalButtonWrapper, ModalTitle } from '../styles';
import { getEditedDataSource, getRecordIndex } from '../utils';
import { renderPipelineForm } from './utils';

function AddNewPipeline({
	isActionType,
	setActionType,
	selectedPipelineData,
	setShowSaveButton,
	setCurrPipelineData,
	currPipelineData,
}: AddNewPipelineProps): JSX.Element {
	const [form] = Form.useForm();
	const { t } = useTranslation('pipeline');
	const { user } = useSelector<AppState, AppReducer>((state) => state.app);
	// const [tagsListData, setTagsListData] = useState<Array<string>>();

	const isEdit = isActionType === 'edit-pipeline';
	const isAdd = isActionType === 'add-pipeline';

	useEffect(() => {
		if (isEdit) {
			// setTagsListData(selectedPipelineData?.tags);
			form.setFieldsValue(selectedPipelineData);
		}
		if (isAdd) {
			// setTagsListData([]);
			form.resetFields();
		}
	}, [form, isEdit, isAdd, selectedPipelineData]);

	const onFinish = (values: PipelineData): void => {
		const newPipeLineData: PipelineData = {
			orderId: currPipelineData.length + 1,
			createdAt: new Date().toISOString(),
			createdBy: user?.name || '',
			name: values.name,
			alias: values.name.replace(/\s/g, ''),
			description: values.description,
			filter: values.filter,
			// tags: tagsListData || [],
			config: [],
			enabled: true,
		};

		if (isEdit && selectedPipelineData) {
			const findRecordIndex = getRecordIndex(
				currPipelineData,
				selectedPipelineData,
				'name',
			);
			const updatedPipelineData: PipelineData = {
				...currPipelineData[findRecordIndex],
				name: values.name,
				description: values.description,
				filter: values.filter,
				// tags: tagsListData || [],
			};

			const editedPipelineData = getEditedDataSource(
				currPipelineData,
				selectedPipelineData,
				'name',
				updatedPipelineData,
			);

			setCurrPipelineData(editedPipelineData);
		}
		if (isAdd) {
			// setTagsListData([]);
			setCurrPipelineData((prevState) => [...prevState, newPipeLineData]);
		}
		setActionType(undefined);
	};

	const onCancelModalHandler = (): void => {
		setActionType(undefined);
	};

	const modalTitle = useMemo(
		(): string =>
			isEdit
				? `${t('edit_pipeline')} : ${selectedPipelineData?.name}`
				: t('create_pipeline'),
		[isEdit, selectedPipelineData?.name, t],
	);

	const onOkModalHandler = useCallback(
		() => setShowSaveButton(ActionMode.Editing),
		[setShowSaveButton],
	);

	const isOpen = useMemo(() => isEdit || isAdd, [isAdd, isEdit]);

	return (
		<Modal
			title={<ModalTitle level={4}>{modalTitle}</ModalTitle>}
			centered
			open={isOpen}
			width={800}
			footer={null}
			onCancel={onCancelModalHandler}
		>
			<Divider plain />
			<Form
				name="add-new-pipeline"
				layout="vertical"
				onFinish={onFinish}
				autoComplete="off"
				form={form}
			>
				{/* {renderPipelineForm(setTagsListData, tagsListData)} */}
				{renderPipelineForm()}
				<Divider plain />
				<Form.Item>
					<ModalButtonWrapper>
						<Button
							key="submit"
							type="primary"
							htmlType="submit"
							onClick={onOkModalHandler}
						>
							{isEdit ? t('update') : t('create')}
						</Button>
						<Button key="cancel" onClick={onCancelModalHandler}>
							{t('cancel')}
						</Button>
					</ModalButtonWrapper>
				</Form.Item>
			</Form>
		</Modal>
	);
}

interface AddNewPipelineProps {
	isActionType: string;
	setActionType: (actionType?: ActionType) => void;
	selectedPipelineData: PipelineData | undefined;
	setShowSaveButton: (actionMode: ActionMode) => void;
	setCurrPipelineData: (
		value: React.SetStateAction<Array<PipelineData>>,
	) => void;
	currPipelineData: Array<PipelineData>;
}

export default AddNewPipeline;
