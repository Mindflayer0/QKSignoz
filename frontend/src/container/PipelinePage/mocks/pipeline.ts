import { PipelineColumn } from '../PipelineListsView/types';

export const configurationVerison = '1.0';

export const pipelineMockData: Array<PipelineColumn> = [
	{
		orderid: 1,
		uuid: '22a588b8-cccc-4a49-94f1-2caa28271315',
		createdAt: '1674205513092137500',
		createdBy: {
			username: 'nitya',
			email: 'nityananda@signoz.io',
		},
		updatedAt: 'Mon Feb 06 2023',
		updatedBy: {
			username: 'nitya',
			email: '',
		},
		version: 'myversion1',
		name: 'Apache common parser',
		alias: 'apachecommonparser',
		enabled: true,
		filter: 'attributes.source == nginx',
		tags: ['server', 'app'],
		operators: [
			{
				id: '1',
				type: 'grok',
				name: 'grok use common',
				output: 'grokcommon',
				pattern: '%{COMMONAPACHELOG}',
				parse_from: 'body',
				parse_to: 'attributes',
			},
			{
				id: '2',
				type: 'move',
				name: 'rename auth',
				output: 'renameauth',
				parse_from: 'attributes.auth',
				parse_to: 'attributes.username',
			},
		],
	},
	{
		orderid: 2,
		uuid: 'f58ce70c-5f41-48bc-949d-a822e2409257',
		version: 'myversion1',
		createdAt: '1674205513092137500',
		createdBy: {
			username: 'nitya',
			email: 'nityananda@signoz.io',
		},
		updatedAt: 'Sun Feb 05 2023',
		updatedBy: {
			username: 'nityananda',
			email: '',
		},
		name: 'myservice trace parser',
		alias: 'myservicetraceparser',
		enabled: true,
		filter: 'attributes.source == myservice',
		tags: ['server', 'app', 'host', 'realse'],
		operators: [
			{
				id: '1',
				type: 'trace_parser',
				name: 'Parse trace details',
				output: 'removeoldxtrace_id',
				trace_id: {
					parse_from: 'attributes.xtrace_id',
				},
				span_id: {
					parse_from: 'attributes.xspan_id',
				},
				trace_flags: {
					parse_from: 'attributes.xtrace_flag',
				},
			},
			{
				id: '2',
				type: 'remove',
				name: 'remove old xtrace_id',
				output: 'removeoldxspan_id',
				field: 'attributes.xtrace_id',
			},
			{
				id: '3',
				type: 'remove',
				name: 'remove old xspan_id',
				output: 'removeoldxtrace_flag',
				field: 'attributes.xspan_id',
			},
			{
				id: '4',
				type: 'remove',
				name: 'remove old xtrace_flag',
				output: 'removeoldxtrace_flag',
				field: 'attributes.xtrace_flag',
			},
		],
	},
];
