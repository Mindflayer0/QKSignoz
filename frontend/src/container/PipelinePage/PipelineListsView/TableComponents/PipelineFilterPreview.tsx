import { queryFilterTags } from 'hooks/queryBuilder/useTag';
import styled from 'styled-components';
import { PipelineData } from 'types/api/pipeline/def';

const FilterCondition = styled.div`
	display: inline;
	padding: 0 0.25em;
`;

const FilterContainer = styled.div`
	display: flex;
	flex-wrap: wrap;
	gap: 0.2rem;
	font-size: 0.75rem;
`;

function PipelineFilterPreview({
	filter,
}: PipelineFilterPreviewProps): JSX.Element {
	return (
		<FilterContainer>
			{queryFilterTags(filter).map((tag) => (
				<FilterCondition key={tag}>{tag}</FilterCondition>
			))}
		</FilterContainer>
	);
}

interface PipelineFilterPreviewProps {
	filter: PipelineData['filter'];
}

export default PipelineFilterPreview;
