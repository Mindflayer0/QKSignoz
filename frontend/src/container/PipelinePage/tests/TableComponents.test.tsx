import { render } from '@testing-library/react';
import { pipelineMockData } from 'container/PipelinePage/mocks/pipeline';
import React from 'react';
import { I18nextProvider } from 'react-i18next';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import i18n from 'ReactI18';
import store from 'store';

import TableComponents from '../PipelineListsView/TableComponents';
import { matchMedia } from './AddNewPipeline.test';

beforeAll(() => {
	matchMedia();
});

describe('PipelinePage container test', () => {
	it('should render TableComponents section', () => {
		const { asFragment } = render(
			<MemoryRouter>
				<Provider store={store}>
					<I18nextProvider i18n={i18n}>
						<TableComponents
							columnKey="id"
							record={pipelineMockData[0].config as never}
						/>
					</I18nextProvider>
				</Provider>
			</MemoryRouter>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
