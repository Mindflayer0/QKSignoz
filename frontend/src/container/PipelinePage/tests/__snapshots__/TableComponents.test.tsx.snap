// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PipelinePage container test should render AddNewPipeline section 1`] = `<DocumentFragment />`;

exports[`PipelinePage container test should render TableComponents section 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #1668DC;
  height: 1rem;
  width: 1rem;
  font-size: 0.75rem;
  line-height: 0.813rem;
  font-weight: 400;
}

<span
    class="ant-avatar ant-avatar-sm ant-avatar-circle c0 css-dev-only-do-not-override-1i536d8"
  >
    <span
      class="ant-avatar-string"
      style="transform: scale(1) translateX(-50%);"
    >
      NaN
    </span>
  </span>
</DocumentFragment>
`;
